src/class/mms-project.mjs

```diff
@@ -303,34 +311,12 @@ export class MmsProject {
 
 		// deletions
 		if(a_deleted.length) {
-			const ds_delete = new stream.PassThrough();
-			const dp_upload = upload(ds_delete, this._endpoint_ref('elements', si_ref, {overwrite:true}), this._gc_req_delete);
-
-			ds_delete.write(/* syntax: json */ `{"elements":[`);
-			let i_element = 0;
-			for(const si_element of a_deleted) {
-				ds_delete.write((i_element++? ',': '')+/* syntax: json */ `\n{"id":"${si_element}"}`);
-			}
-			ds_delete.end(/* syntax: json */ `\n]}`);
-
-			await once(ds_delete, 'finish');
-			await dp_upload;
+			await this.upload_passthrough(a_deleted, si_ref, this._gc_req_delete);
 		}
 
 		// additions
 		if(a_added.length) {
-			const ds_add = new stream.PassThrough();
-			const dp_upload = upload(ds_add, this._endpoint_ref('elements', si_ref, {overwrite:true}), this._gc_req_post);
-
-			ds_add.write(/* syntax: json */ `{"elements":[`);
-			let i_element = 0;
-			for(const g_element of a_added) {
-				ds_add.write((i_element++? ',': '')+'\n'+JSON.stringify(g_element));
-			}
-			ds_add.end(/* syntax: json */ `\n]}`);
-
-			await once(ds_add, 'finish');
-			await dp_upload;
+			await this.upload_passthrough(a_added, si_ref, this._gc_req_post);
```

Comment:
this change seems to miss out on passing the whole stringified element as it will only pass the id of each element to add without its body